### PR TITLE
Fix extended names in on_trait_change lists.

### PR DESCRIPTION
**Pull request by mdickinson**
Fix for #403.

- [x] Step 1: add failing test.
- [x] Step 2: fix the test!

### CODE REVIEW

File: traits/has_traits.py [hunk 1]
```diff
@@ -2562,8 +2562,15 @@ def on_trait_change ( self, handler, name = None, remove = False,
...
             for name_i in name:
-                self.on_trait_change( handler, name_i, remove, dispatch,
-                                      priority, target )
+                self.on_trait_change(
+                    handler,
+                    name=name_i,
+                    remove=remove,
+                    dispatch=dispatch,
+                    priority=priority,
+                    deferred=deferred,
```
Comment (corranwebster) [inline traits/has_traits.py]: I think this is not meant to be here?

Comment (corranwebster) [reply traits/has_traits.py]: You're right - wasn't aware of the deferred option and it isn't present in the call on line 2555 above and appears not to be documented anywhere.  In fact deferred seems to do nothing. Is this a dead part of the API?

Comment (mdickinson) [reply traits/has_traits.py]: The line 2555 call is to a different method: `_on_trait_change`, not `on_trait_change`.

Don't know what `deferred` is *supposed* to do; I can open a separate issue for that.

Comment (mdickinson) [reply traits/has_traits.py]: Looks like `deferred` is tied into the `post_init` machinery. I'd be nervous about messing with it in this PR.

Comment (corranwebster) [reply traits/has_traits.py]: Hmmm... it is used below.  Apologies for the noise.  Clearly I need to work out up what it actually does.

**Pull request merged**